python/lsst/ctrl/bps/wms/panda/panda_service.py

```diff
@@ -0,0 +1,151 @@
+        for task in workflow.generated_tasks:
+            work = DomaLSSTWork(
+                executable=self.add_decoder_prefix(task.executable),
+                primary_input_collection={'scope': 'pseudo_dataset', 'name': 'pseudo_input_collection#1'},
+                output_collections=[{'scope': 'pseudo_dataset', 'name': 'pseudo_output_collection#1'}],
+                log_collections=[], dependency_map=task.dependencies,
+                task_name=task.name,
+                task_queue=task.queue)
+            idds_client_workflow.add_work(work)
+        host = get_rest_host()
+        idds_workflow_manager = IDDS_workflow_manager(host=host)
+        request_id = idds_workflow_manager.submit(idds_client_workflow)
+        _LOG.info("Submitted into iDDs with request id=%i", request_id)
+        workflow.run_id = request_id
+
+    def report(self, wms_workflow_id=None, user=None, hist=0, pass_thru=None):
+        """Return run information based upon given constraints.
+
+        Parameters
+        ----------
+        wms_workflow_id : `int` or `str`
+            Limit to specific run based on id.
+        user : `str`
+            Limit results to runs for this user.
+        hist : `float`
+            Limit history search to this many days.
+        pass_thru : `str`
+            Constraints to pass through to HTCondor.
+
+        Returns
+        -------
+        runs : `dict` of `~lsst.ctrl.bps.wms_service.WmsRunReport`
+            Information about runs from given job information.
+        message : `str`
+            Extra message for report command to print.  This could be pointers to documentation or
+            to WMS specific commands.
+        """
+        message = ""
+        run_reports = None
+        return run_reports, message
+
+
+class PanDABPSWMSWorkflow(BaseWmsWorkflow):
+    def __init__(self, name, config=None):
+        super().__init__(name, config)
+        self.generated_tasks = None
+
+    @classmethod
+    def from_generic_workflow(cls, config, generic_workflow, out_prefix, service_class):
+        idds_workflow = cls(generic_workflow.name, config)
+        workflow_generator = IDDSWorkflowGenerator(generic_workflow, config)
+        idds_workflow.generated_tasks = workflow_generator.define_tasks()
+        for task in idds_workflow.generated_tasks:
+            cloud_prefix = config.get('bucket') + '/' + \
+                config.get('payload_folder') + '/' + config.get('workflowName') + '/'
+            cls.copy_pickles_in_cloud(task.local_pfns, cloud_prefix)
+        _LOG.debug("panda dag attribs %s", generic_workflow.run_attrs)
+        return idds_workflow
+
+    @classmethod
+    def copy_pickles_in_cloud(cls, local_pfns, cloud_prefix):
+        for src_path in local_pfns:
+            src = ButlerURI(src_path)
+            target = ButlerURI(cloud_prefix + '/' + os.path.basename(src_path))
+            target.transfer_from(src, transfer="copy")
+
+    def write(self, out_prefix):
```

Comment:
This doesn't actually write anything because `from_generic_workflow` does all the writing?

Reply:
If not implementing, should this just be removed (after removing call from prepare)?